src/main/flight/imu.c

```diff
@@ -609,6 +606,25 @@ static void imuCalculateEstimatedAttitude(timeUs_t currentTimeUs)
 }
 #else
 
+#if defined(USE_GPS)
+static void isGpsHeadingUsable(float groundspeedGain, float imuCourseError, float dt)
+{
+    // ** ATTEMPT TO DETECT SUCCESSFUL IMU ORIENTATION TO GPS **
+    static float gpsHeadingTruth = 0;
+    // groundspeedGain can be 5.0 in clean forward flight, up to 10.0 max
+    // fabsf(imuCourseError) is 0 when headings are aligned, 1 when 90 degrees error or worse
+    // accumulate 'points' based on alignment and likelihood of accumulation being good
+    gpsHeadingTruth += fmaxf(groundspeedGain - fabsf(imuCourseError), 0.0f) * dt;
+    // recenter at 2.5s time constant
+    // TODO: intent is to match IMU time constant, approximately, but I don't exactly know how to do that
+    gpsHeadingTruth -= 0.4 * dt * gpsHeadingTruth; 
+    // if we accumulate enough 'points' over time, the IMU probably is OK
+    // will need to reaccumulate after a disarm (will be retained partly for very brief disarms)
+    canUseGPSHeading = !canUseGPSHeading && (gpsHeadingTruth > 2.0f);
+    // latches true; reset on disarm; used to stop position hold without a suitable GPS heading, if needed
```

Comment:
Does not latch, resets canUseGPSHeading each other cycle

Reply:
The intent was that, after power up, when the user flies straight ahead for long enough, the gpsHeadingTruth rises, and when it exceeds the threshold, canUseGPSHeading is set true, and stays true.

Modified the code and comments slightly to make this explicit.

I am yet to test this, and will report back when I do.

Reply:
when `canUseGPSHeading` is true, canUseGPSHeading = !true && (anything)

Maybe you wanted `canUseGPSHeading = canUseGPSHeading || (gpsHeadingTruth > 2.0f);` ?